homeassistant/components/unifi/update.py

```diff
@@ -0,0 +1,139 @@
+"""Update entities for Ubiquiti network devices."""
+from __future__ import annotations
+
+import logging
+
+from homeassistant.components.update import (
+    DOMAIN,
+    UpdateDeviceClass,
+    UpdateEntity,
+    UpdateEntityFeature,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_NAME
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.device_registry import CONNECTION_NETWORK_MAC
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import ATTR_MANUFACTURER, DOMAIN as UNIFI_DOMAIN
+from .unifi_entity_base import UniFiBase
+
+LOGGER = logging.getLogger(__name__)
+
+DEVICE_UPDATE = "device_update"
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up update entities for UniFi Network integration."""
+    controller = hass.data[UNIFI_DOMAIN][config_entry.entry_id]
+    controller.entities[DOMAIN] = {DEVICE_UPDATE: set()}
+
+    @callback
+    def items_added(
+        clients: set = controller.api.clients, devices: set = controller.api.devices
+    ) -> None:
+        """Add device update entities if enabled."""
+        if controller.option_device_updates:
+            add_device_update_entities(controller, async_add_entities, devices)
+
+    for signal in (controller.signal_update, controller.signal_options_update):
+        config_entry.async_on_unload(
+            async_dispatcher_connect(hass, signal, items_added)
+        )
+
+    items_added()
+
+
+@callback
+def add_device_update_entities(controller, async_add_entities, devices):
+    """Add new device update entities from the controller."""
+    entities = []
+
+    for mac in devices:
+        if mac in controller.entities[DOMAIN][UniFiDeviceUpdateEntity.TYPE]:
+            continue
+
+        device = controller.api.devices[mac]
+        entities.append(UniFiDeviceUpdateEntity(device, controller))
+
+    if entities:
+        async_add_entities(entities)
+
+
+class UniFiDeviceUpdateEntity(UniFiBase, UpdateEntity):
+    """Update entity for a UniFi network infrastructure device."""
+
+    DOMAIN = DOMAIN
+    TYPE = DEVICE_UPDATE
```

Comment:
I think this does not need to be a primary entity
```suggestion
    TYPE = DEVICE_UPDATE
    __attr_entity_category = EntityCategory.DIAGNOSTIC
```

Reply:
`DIAGNOSTIC` is already the default at https://github.com/home-assistant/core/blob/49491bcda55125369319fe1f0b6036ddfc4a2a2a/homeassistant/components/update/__init__.py#L215-L224

So I don't think it is needed to set it here also?

Reply:
Perfect!